src/webgpu/api/validation/encoding/cmds/render/setPipeline.spec.ts

```diff
@@ -30,14 +30,14 @@ Tests setPipeline should generate an error iff using an 'invalid' pipeline.
 
 g.test('pipeline,device_mismatch')
   .desc('Tests setPipeline cannot be called with a render pipeline created from another device')
-  .paramsSubcasesOnly(kRenderEncodeTypeParams.combine('mismatched', [true, false]))
-  .fn(async t => {
-    const { encoderType, mismatched } = t.params;
-
-    if (mismatched) {
+  .params(u => u.combine('mismatched', [true, false]).combineWithParams(kRenderEncodeTypeParams))
```

Comment:
intended to have a .beginSubcases() in here?

Also for all of the device_mismatch tests throughout the test suite - I think we could leave the `mismatched` param in the subcases and selectMismatchedDeviceOrSkipTestCase unconditionally. It doesn't hurt to grab an extra device unconditionally for just these tests.

Reply:
done. most of these can go back to paramsSubcasesOnly then